# Juglandaceae
<p style="font-size: 12pt; -webkit-user-drag: none; margin: 0cm 0cm 0cm 21pt; text-align: justify; text-indent: -21pt; font-family: &quot;Times New Roman&quot;, serif;"><span lang="EN-US" style="-webkit-user-drag: none; font-family: Symbol;"><span style="-webkit-user-drag: none;">·<span style="-webkit-user-drag: none; font-stretch: normal; font-size: 7pt; line-height: normal; font-family: &quot;Times New Roman&quot;;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></span></span><span lang="EN-US" style="-webkit-user-drag: none;">Blast/: The scripts were used to get BLASTP hits and reciprocal best hits (RBHs).</span></p><p style="font-size: 12pt; -webkit-user-drag: none; margin: 0cm 0cm 0cm 21pt; text-align: justify; text-indent: -21pt; font-family: &quot;Times New Roman&quot;, serif;"><span lang="EN-US" style="-webkit-user-drag: none; font-family: Symbol;"><span style="-webkit-user-drag: none;">·<span style="-webkit-user-drag: none; font-stretch: normal; font-size: 7pt; line-height: normal; font-family: &quot;Times New Roman&quot;;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></span></span><span lang="EN-US" style="-webkit-user-drag: none;">Gene-content/: The scripts were used to get cluster information of each orthogroups.</span></p><p style="font-size: 12pt; -webkit-user-drag: none; margin: 0cm 0cm 0cm 21pt; text-align: justify; text-indent: -21pt; font-family: &quot;Times New Roman&quot;, serif;"><span lang="EN-US" style="-webkit-user-drag: none; font-family: Symbol;"><span style="-webkit-user-drag: none;">·<span style="-webkit-user-drag: none; font-stretch: normal; font-size: 7pt; line-height: normal; font-family: &quot;Times New Roman&quot;;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></span></span><span lang="EN-US" style="-webkit-user-drag: none;">KaKs_calculator/: The scripts were used to compute Ka, Ks, Ka/Ks for collinear gene pairs, and got the median Ks value for each collinear block.</span></p><p style="font-size: 12pt; -webkit-user-drag: none; margin: 0cm 0cm 0cm 21pt; text-align: justify; text-indent: -21pt; font-family: &quot;Times New Roman&quot;, serif;"><span lang="EN-US" style="-webkit-user-drag: none; font-family: Symbol;"><span style="-webkit-user-drag: none;">·<span style="-webkit-user-drag: none; font-stretch: normal; font-size: 7pt; line-height: normal; font-family: &quot;Times New Roman&quot;;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></span></span><span lang="EN-US" style="-webkit-user-drag: none;">KEGG-DNA-repair/: The scripts were used to get KEGG Orthology (KO) annotation.</span></p><p style="font-size: 12pt; -webkit-user-drag: none; margin: 0cm 0cm 0cm 21pt; text-align: justify; text-indent: -21pt; font-family: &quot;Times New Roman&quot;, serif;"><span lang="EN-US" style="-webkit-user-drag: none; font-family: Symbol;"><span style="-webkit-user-drag: none;">·<span style="-webkit-user-drag: none; font-stretch: normal; font-size: 7pt; line-height: normal; font-family: &quot;Times New Roman&quot;;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></span></span><span lang="EN-US" style="-webkit-user-drag: none;">Share-WGD/: The scripts were used to detect homologous quartets, constructed gene trees and counted the number of each topology.</span></p><p style="-webkit-user-drag: none; font-family: Arial;"></p><p style="font-size: 12pt; -webkit-user-drag: none; margin: 0cm 0cm 0cm 21pt; text-align: justify; text-indent: -21pt; font-family: &quot;Times New Roman&quot;, serif;"><span lang="EN-US" style="-webkit-user-drag: none; font-family: Symbol;">·<span style="-webkit-user-drag: none; font-stretch: normal; font-size: 7pt; line-height: normal; font-family: &quot;Times New Roman&quot;;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</span></span><span lang="EN-US" style="-webkit-user-drag: none;">Plot_script/: The R scripts were used to plot the distribution of Ks value.</span></p>
<div><font color="#24292f" face="-apple-system, system-ui, Segoe UI, Noto Sans, Helvetica, Arial, sans-serif, Apple Color Emoji, Segoe UI Emoji" size="3">#Citation</font></div><div>Ding, Y.-M., Pang, X.-X., Cao, Y., Zhang, W.-P., Renner, S. S., et al. 2023. Genome structure-based Juglandaceae phylogenies contradict alignment-based phylogenies and substitution rates vary with DNA repair genes.&nbsp;<i>Nature Communications.</i><b>14</b>:617.</div>
